nsv13/game/custom_turfs.dm

```diff
@@ -4,28 +4,52 @@
 	sheet_type = /obj/item/stack/sheet/durasteel
 	var/connect_universally = TRUE //Connect to every subtype of the walls?
 
+/obj/structure/falsewall/durasteel
+	icon = 'nsv13/icons/turf/interior_wall.dmi'
+	name = "Durasteel hull"
+	desc = "A large hull segment designed to create vessels and structures capable of supporting life in even the most hazardous places."
+	legacy_smooth = TRUE //Override /tg/ iconsmooths
+	smooth = TRUE
+	walltype = /turf/closed/wall/ship
+	canSmoothWith = list(/turf/closed/wall/ship,/turf/closed/wall/r_wall/ship,/obj/machinery/door,/obj/structure/window, /obj/structure/falsewall/duranium, /obj/structure/falsewall/durasteel)
+	mineral = /obj/item/stack/sheet/durasteel
+	var/connect_universally = TRUE //Connect to every subtype of the walls?
+
 /turf/closed/wall/r_wall/ship
 	icon = 'nsv13/icons/turf/reinforced_wall.dmi'
 	name = "Duranium hull"
 	desc = "A large hull segment designed to create vessels and structures capable of supporting life in even the most hazardous places."
 	legacy_smooth = TRUE //Override /tg/ iconsmooths
 	smooth = TRUE
-	canSmoothWith = list(/turf/closed/wall/ship,/obj/machinery/door,/obj/structure/window,/turf/closed/wall/r_wall/ship)
+	canSmoothWith = list(/turf/closed/wall/ship,/obj/machinery/door,/obj/structure/window,/turf/closed/wall/r_wall/ship, /obj/structure/falsewall/duranium, /obj/structure/falsewall/durasteel)
 	sheet_type = /obj/item/stack/sheet/duranium
+	sheet_amount = 2
+	girder_type = /obj/structure/girder
 
-/obj/structure/girder/attackby(obj/item/W, mob/user, params) //Time to add support for our walls..
-	var/obj/item/stack/sheet/S = W
-	if(istype(W, /obj/item/stack/sheet))
+/obj/structure/falsewall/duranium
+	icon = 'nsv13/icons/turf/reinforced_wall.dmi'
+	name = "Duranium hull"
+	desc = "A large hull segment designed to create vessels and structures capable of supporting life in even the most hazardous places."
+	legacy_smooth = TRUE //Override /tg/ iconsmooths
+	smooth = TRUE
+	walltype = /turf/closed/wall/r_wall/ship
+	canSmoothWith = list(/turf/closed/wall/ship,/obj/machinery/door,/obj/structure/window,/turf/closed/wall/r_wall/ship, /obj/structure/falsewall/duranium, /obj/structure/falsewall/durasteel)
+	mineral = /obj/item/stack/sheet/duranium
+
+/obj/structure/girder/proc/try_nsv_walls(obj/item/stack/sheet/S, mob/user)
+	if(!S.turf_type) //Let the girder code handle it
+		return
+
+	if (state != GIRDER_DISPLACED) //Build regular wall
 		if(S.get_amount() < 2)
-			to_chat(user, "<span class='warning'>You need two sheets of [W] to finish a wall!</span>")
-			return
-		if(!S.turf_type)
-			to_chat(user, "<span class='warning'>You can't build anything with [W].</span>")
+			to_chat(user, "<span class='warning'>You need two sheets of [S] to finish a wall!</span>")
 			return
+
+		to_chat(user, "<span class='notice'>You start plating the wall with [S]...</span>")
 		if (do_after(user, 40, target = src))
 			if(S.get_amount() < 2)
```

Comment:
you should add in the to_chat warning here that you need extra sheets I believe.

Reply:
This is the success path. `S.use(2)` needs to be called as well to actually use the material from the stack.

Reply:
As well as actual wall creation unless I'm missing something.

Reply:
Added warning.
Wall creation was just collapsed by github because it hadn't changed. Visible now that I changed the return.